index.js

```diff
@@ -54,6 +65,21 @@ module.exports = function (src, dest, opts) {
 		return new Promise(function pipeToDest(resolve, reject) {
 			var write = fs.createWriteStream(dest, {flags: opts.overwrite ? 'w' : 'wx'});
 
+			read.on('data', function () {
+				if (size === 0) {
+					return;
+				}
+				var written = write.bytesWritten;
+				var percent = written / size;
+				progressEmitter.emit('progress', {
+					src: src,
+					dest: dest,
+					size: size,
+					written: written,
+					percent: percent
```

Comment:
What if user wants a better precision?
Say if you have a progress bar with value in pixels, and you calc `done` value like
```js
const done = progress * fullWidth;
```
Given that file can be big, you can have multiple progress events that rounds for same `0.45` value, making it stay at same position during multiple progress events.